testhelpers/testhelpers.go

```diff
@@ -184,12 +184,26 @@ func PushImage(dockerCli dockercli.CommonAPIClient, ref string) error {
 	return rc.Close()
 }
 
-func HTTPGetE(url string) (string, error) {
-	resp, err := http.DefaultClient.Get(url)
+func HTTPGetE(url string, headers ...map[string]string) (string, error) {
+	client := http.DefaultClient
+
+	request, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "making new request")
+	}
+
+	if len(headers) > 0 {
+		for key, val := range headers[0] {
+			request.Header.Set(key, val)
+		}
+	}
+
```

Comment:
```suggestion
	for key, val := range headers {
		request.Header.Set(key, val)
	}

```

Reply:
This won't compile because `headers` is a list of maps. I did it this way (variadic headers) to avoid breaking current consumers of `HTTPGetE` - but it looks like the only usage within the repo is within docker_registry.go. Should I make `headers` a required argument?

Reply:
About to push some code that has this change (making it required) - LMK if you want me to put it back :)

Reply:
The changes look good. Sorry I assumed, rather than checking the type of `headers`. In any case in seems wrong to except variadic headers and then only respect the first one. Making it required is better imo.